pkg/compact/compact_test.go

```diff
@@ -522,29 +521,44 @@ func TestCompactProgressCalculate(t *testing.T) {
 			return
 		}
 	}
-
 }
 
 func TestDownsampleProgressCalculate(t *testing.T) {
 	reg := prometheus.NewRegistry()
 	unRegisterer := &receive.UnRegisterer{Registerer: reg}
-	extLabels := labels.FromMap(map[string]string{"a": "1", "b": "2"})
 	logger := log.NewNopLogger()
+	type groupedResult map[string]float64
+
+	// pre calculating group keys
+	keys := make([]string, 3)
+	m := make([]metadata.Meta, 3)
+	m[0].Thanos.Labels = map[string]string{"a": "1"}
+	m[0].Thanos.Downsample.Resolution = downsample.ResLevel0
+	m[1].Thanos.Labels = map[string]string{"b": "2"}
+	m[1].Thanos.Downsample.Resolution = downsample.ResLevel1
+	m[2].Thanos.Labels = map[string]string{"a": "1", "b": "2"}
+	m[2].Thanos.Downsample.Resolution = downsample.ResLevel2
+	for ind, meta := range m {
+		keys[ind] = DefaultGroupKey(meta.Thanos)
+	}
+
+	var bkt objstore.Bucket
+	temp := prometheus.NewCounter(prometheus.CounterOpts{})
+	grouper := NewDefaultGrouper(logger, bkt, false, false, reg, temp, temp, temp, "")
 
 	for _, tcase := range []struct {
 		testName string
 		input    []*metadata.Meta
-		expected float64
+		expected groupedResult
 	}{
 		{
-			// This test case has 1 block to be downsampled out of 2 since for the second block, the difference between MinTime and MaxTime is less than the acceptable threshold, DownsampleRange0
-			testName: "minTime_maxTime_diff_threshold_test",
+			testName: "first_test",
```

Comment:
The test case name doesn't mean anything. Can you update all the names?

Reply:
This is already fixed in a previous commit.

Reply:
I don't think these are fixed.

Reply:
Thanks, fixed this for downsampling in a recent commit. Will fix for compact progress next.

Reply:
Fixed for both tests now.